src/main/java/teammates/common/datatransfer/questions/FeedbackRankRecipientsQuestionDetails.java

```diff
@@ -285,6 +310,60 @@ public String getQuestionResultStatisticsCsv(
         return normalisedRankOfResponse;
     }
 
+    /**
+     * Returns list of responses excluding responses given to self.
+     *
+     * @param responses a list of responses
+     * @return list of responses excluding self given responses
+     */
+    private List<FeedbackResponseAttributes> getResponsesExcludingSelf(List<FeedbackResponseAttributes> responses) {
+        List<FeedbackResponseAttributes> responsesExcludingSelf = new ArrayList<FeedbackResponseAttributes>();
+        for (FeedbackResponseAttributes response : responses) {
+            if (!response.giver.equalsIgnoreCase(response.recipient)) {
+                responsesExcludingSelf.add(response);
+            }
+        }
+        return responsesExcludingSelf;
+    }
+
+    /**
+     * Returns the average excluding self response text.
+     * Displays a dash if the user has only self response.
+     *
+     * @param df decimal format
+     * @param recipientRanksExcludingSelf list of recipient ranks excluding self response
+     * @param recipientKey recipient for which average is to be calculated
```

Comment:
Should this be `recipientName`? The fact that it is used as a key for some key-value pairing should not make its name `recipientKey`.